packages/remirror__react-components/src/find-replace/find-replace-component.tsx

```diff
@@ -0,0 +1,226 @@
+import { Box, Button, ButtonGroup, IconButton, InputAdornment, OutlinedInput } from '@mui/material';
+import React, { FC, useEffect } from 'react';
+import { useCommands, useHelpers } from '@remirror/react-core';
+
+import { Icon } from '../icons';
+import { MdiFormatLetterCase } from './letter-icon';
+
+function useSearch() {
+  const helpers = useHelpers();
+  const commands = useCommands();
+  const [searchTerm, setSearchTerm] = React.useState('');
+  const [currIndex, setCurrIndex] = React.useState(-1);
+  const [total, setTotal] = React.useState(0);
+  const [caseSensitive, setCaseSensitive] = React.useState(false);
+  const [replacement, setReplacement] = React.useState('');
+
+  const search = (index: number) => {
```

Comment:
Maybe add `useCallback` to these functions to avoid recreating them on each rerender?